torch/fx/experimental/Partitioner.py

```diff
@@ -245,18 +259,47 @@ def dump_partition_DAG(self, module_with_submodules: GraphModule) -> DAG:
                 output_nodes = list(node.users)
             else:
                 output_nodes = [node]
-            dag.create_node(node, list(input_nodes), output_nodes)
+            dag.create_node(node, list(input_nodes), output_nodes, [])
+        # Assign logical devices to each node
```

Comment:
Assigning logicalDevices can be a global operation, there's no need to hide it behind the sparsenn flag.
The DAG returned should alway have the same fields and format, no matter which partitioning scheme is used.

Reply:
This may mean logicalDeviceID is something you'll want to store in Partition so you don't need special logic in dump_partition_DAG